File: source/bindings/js/src/sdk/TranslationTextResultCanceledEventArgs.ts
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

import {
    CancellationReason,
    TranslationTextResult,
} from "./Exports";
/**
 * Define payload of speech recognition canceled result events.
 */
export class TranslationTextResultCanceledEventArgs {

    private privResult: TranslationTextResult;
    private privSessionId: string;
    private privCancelReason: CancellationReason;
    private privErrorDetails: string;

    public constructor(sessionid: string, cancellationReason: CancellationReason, errorDetails: string, result: TranslationTextResult) {
        this.privCancelReason = cancellationReason;
        this.privErrorDetails = errorDetails;
        this.privResult = result;
        this.privSessionId = sessionid;
    }

    /**
     * Specifies the recognition result.
     * @return the recognition result.
     */
    public get result(): TranslationTextResult {
        return this.privResult;
    }

    /**
     * Specifies the session identifier.
     * @return the session identifier.
     */
    public get sessionId(): string {
        return this.privSessionId;
    }

    /**
     * The reason the recognition was canceled.
     * @return Specifies the reason canceled.
     */
    public get reason(): CancellationReason {
        return this.privCancelReason;
    }

    /**
     * In case of an unsuccessful recognition, provides a details of why the occurred error.
     * This field is only filled-out if the reason canceled (@see getReason) is set to Error.
     * @return A String that represents the error details.
     */
    public get errorDetails(): string {
        return this.privErrorDetails;
    }
}

File: source/bindings/java/com/microsoft/cognitiveservices/speech/translation/TranslationTextResultCanceledEventArgs.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech.translation;

import com.microsoft.cognitiveservices.speech.CancellationReason;
import com.microsoft.cognitiveservices.speech.util.Contracts;

/**
 * Defines payload of translation canceled events.
 */
public final class TranslationTextResultCanceledEventArgs extends TranslationTextResultEventArgs {

    TranslationTextResultCanceledEventArgs(com.microsoft.cognitiveservices.speech.internal.TranslationTextResultCanceledEventArgs e) {
        super(e);
        
        Contracts.throwIfNull(e, "e");
        this._eventArgImpl = e;

        com.microsoft.cognitiveservices.speech.internal.CancellationDetails cancellation = e.GetCancellationDetails();
        this._cancellationReason  = com.microsoft.cognitiveservices.speech.CancellationReason.values()[cancellation.getReason().swigValue() - 1]; // Native CancellationReason enum starts at 1!!
        this._errorDetails = cancellation.getErrorDetails();
    }

    /**
     * The reason the recognition was canceled.
     * @return Specifies the reason canceled.
     */
    public CancellationReason getReason() {
        return this._cancellationReason ;
    }

    /**
     * In case of an unsuccessful recognition, provides a details of why the occurred error.
     * This field is only filled-out if the reason canceled (@see getReason) is set to Error.
     * @return A String that represents the error details.
     */
    public String getErrorDetails() {
        return this._errorDetails;
    }

    /**
     * Returns a String that represents the speech recognition canceled event args.
     * @return A String that represents the speech recognition canceled event args.
     */
    @Override
    public String toString() {
        return "SessionId:" + this.getSessionId() +
                " ResultId:" + this.getResult().getResultId() +
                " CancellationReason:" + _cancellationReason  +
                " Recognized text:<" + _errorDetails +
                ">.";
    }

    @SuppressWarnings("unused")
    private com.microsoft.cognitiveservices.speech.internal.TranslationTextResultCanceledEventArgs _eventArgImpl;
    private CancellationReason _cancellationReason ;
    private String _errorDetails;
}
